src/commonMain/kotlin/exchange.dydx.abacus/processor/router/skip/SkipChainResourceProcessor.kt

```diff
@@ -10,7 +11,7 @@ internal class SkipChainResourceProcessor(private val parser: ParserProtocol) {
     ): TransferInputChainResource {
         return TransferInputChainResource(
             chainName = parser.asString(payload["chain_name"]),
-            rpc = parser.asString(payload["rpc"]),
+            rpc = parser.asString(payload["chain_id"])?.let { RpcConfigs.chainIdToRpcMap[it]?.rpcUrl },
```

Comment:
@prashanDYDX wondering if there is a better place to put this than the processor. Doesn't feel right to look up a mapping in the processor since the mapping is part of the overall state. I feel like a processor should be declarative and not depend on context state :/

Reply:
you could remove rpc as a property entirely here, replace it with chain id and do the look ups at usage sites. that way if the rpc configs update we don't have an outdated endpoints hanging around.

Reply:
> at usage sites
meaning client?

> that way if the rpc configs update we don't have an outdated endpoints hanging around
the rpc endpoints are updated only when chains are updated. I think by fixing the race condition Rui pointed out, they will either both be stale or both be up-to-date. Maybe I'm misunderstanding though

Reply:
yeah basically i was suggesting that you don't even need to fix the race condition if you let callers do the lookup themselves

Reply:
i guess i still lean towards doing this in abacus. Otherwise, 
1. consumer context of the rpc config has to understand the availability ordering of the rpc config data
2. more logic on client-side

Eventually we will probably have a similar scenario, so maybe doing it here can serve as an forward example of how to handle concurrent requests/coroutines?

maybe we can chat live about it.